tests/tck-build-logic/src/main/java/org/graalvm/internal/tck/ContributionTask.java

```diff
@@ -0,0 +1,586 @@
+            if (predefinedClassesConfig.get(0).type().equalsIgnoreCase("agent-extracted") && predefinedClassesConfig.get(0).classes().isEmpty()) {
+                removeConfigFile(predefinedClassesConfigPath, CONFIG_FILES.PREDEFINED_CLASSES, remainingFiles);
+            }
+        }
+
+        Path agentExtractedPredefinedClasses = metadataDirectory.resolve("agent-extracted-predefined-classes");
+        if (Files.exists(agentExtractedPredefinedClasses)) {
+            File[] extractedPredefinedClasses = new File(agentExtractedPredefinedClasses.toUri()).listFiles();
+            if (extractedPredefinedClasses == null || extractedPredefinedClasses.length == 0) {
+                InteractiveTaskUtils.printUserInfo("Removing empty: agent-extracted-predefined-classes");
+                invokeCommand("rm -r " + agentExtractedPredefinedClasses, "Cannot delete empty config file: " + agentExtractedPredefinedClasses);
+            }
+        }
+
+        trimIndexFile(indexFile, remainingFiles);
+    }
+
+    private void removeConfigFile(Path path, CONFIG_FILES file, List<CONFIG_FILES> remainingFiles) {
+        InteractiveTaskUtils.printUserInfo("Removing empty: " + file.get());
+        invokeCommand("rm " + path, "Cannot delete empty config file: " + path);
+        remainingFiles.remove(file);
+    }
+
+    private void trimIndexFile(Path index, List<CONFIG_FILES> remainingFiles) throws IOException {
+        InteractiveTaskUtils.printUserInfo("Removing sufficient entries from: " + index);
+        ConfigurationStringBuilder sb = new ConfigurationStringBuilder();
+        sb.openArray().newLine();
+        sb.indent();
+        for (int i = 0; i < remainingFiles.size(); i++) {
+            sb.quote(remainingFiles.get(i).get());
+
+            if (i != remainingFiles.size() - 1) {
+                sb.concat();
+            }
+
+            sb.newLine();
+        }
+
+        sb.unindent();
+        sb.closeArray();
+
+        writeToFile(index, sb.toString(), StandardOpenOption.TRUNCATE_EXISTING);
+    }
+
+    private boolean shouldCreatePullRequest() {
+        ContributingQuestion question = questions.get("shouldCreatePullRequest");
+        return InteractiveTaskUtils.askYesNoQuestion(question.question(), question.help(), true);
+    }
+
+    private void createPullRequest(String branch) {
+        InteractiveTaskUtils.printUserInfo("Creating new branch: " + branch);
+        invokeCommand("git switch -C " + branch, "Cannot create a new branch");
+
+        InteractiveTaskUtils.printUserInfo("Staging changes");
+        invokeCommand("git add .", "Cannot add changes");
+
+        InteractiveTaskUtils.printUserInfo("Committing changes");
+        invokeCommand("git", List.of("commit", "-m", "Add metadata for " + coordinates), "Cannot commit changes", null);
+
+        InteractiveTaskUtils.printUserInfo("Pushing changes");
+        invokeCommand("git push origin " + branch, "Cannot push to origin");
```

Comment:
This assumes you are working on a fork, right? Because that would fail if a user just cloned the reachability repo and doesn't have write privileges to the repo.